packages/agents/src/tests/mcp/transports/auto.test.ts

```diff
@@ -71,14 +71,18 @@ describe("Auto Transport Mode", () => {
       expect(connection.connectionState).toBe("failed");
     });
 
-    it("should fail when asking for an incorrect transport type", async () => {
+    it("should succeed when using different transport than URL endpoint suggests", async () => {
+      // This test verifies that URL normalization allows transport flexibility
       const connection = await initializeMCPClientConnection(
-        "http://example.com/mcp",
-        "sse"
+        "http://example.com/mcp", // URL suggests streamable-http
+        "sse" // but we explicitly request SSE
       );
 
-      await expect(connection.init()).rejects.toThrow();
-      expect(connection.connectionState).toBe("failed");
+      await connection.init();
+
+      // Should succeed because URL normalization strips /mcp and tries /sse
+      expect(connection.connectionState).toBe("ready");
+      expect(connection.tools).toBeDefined();
```

Comment:
This is the main behavioural change in this PR that I'm reconsidering. Everything else is net positive.

Reply:
And this is not a _necessary_ change, but keeping it this way makes it harder to shoot yourself in the foot as a user. Our auto mode actually works pretty well at this point, and we can just try both transports one after the other.

But I'm wondering if there are valid reasons to trust that explicit > implicit:
1. Explicit gives more control to devs over the upgrade path for spec changes in future
2. More control over error scenarios too